lucene/misc/src/java/org/apache/lucene/misc/store/DirectIODirectory.java

```diff
@@ -307,7 +314,14 @@ public void close() throws IOException {
 
     @Override
     public long getFilePointer() {
-      return filePos + buffer.position();
+      long filePointer = filePos + buffer.position();
+
+      // opening the input and immediately calling getFilePointer without calling readX (and thus refill) first,
+      // will result in negative value equal to bufferSize being returned,
+      // due to the initialization method filePos = -bufferSize used in constructor.
+      assert filePointer == -buffer.capacity() || filePointer >= 0 :
+        "filePointer should either be initial value equal to negative buffer capacity, or larger than or equal to 0";
+      return Math.max(filePointer, 0);
```

Comment:
Thanks, this was driving me crazy when I tried to fix this (I gave up after some beers deep night).